Source/town.cpp

```diff
@@ -1279,13 +1279,13 @@ void SetTownMicros()
 	}
 
 	if (zoomflag) {
-		scr_pix_width = 640;
+		scr_pix_width = SCREEN_WIDTH;
 		scr_pix_height = VIEWPORT_HEIGHT;
 		dword_5C2FF8 = 10;
 		dword_5C2FFC = 11;
 	} else {
-		scr_pix_width = 384;
-		scr_pix_height = 224;
+		scr_pix_width = SCREEN_WIDTH / 2 + 64;

```

Comment:
Is the `64` related to `SCREEN_X`?

Reply:
I doubt it since the value is simple `scr_pix_width = SCREEN_WIDTH;` when not zoomed.

Reply:
actually it's neither. the values are an absolute value for the resolution, divided by pixel count for number of arches. this is 64x64 (two arches) for width and 32x32 for height.
```c
scr_pix_width = ZOOM_WIDTH;
scr_pix_height = ZOOM_HEIGHT;
dword_5C2FF8 = ZOOM_WIDTH / 64;
dword_5C2FFC = ZOOM_HEIGHT / 32;
```

Reply:
Also these values are completely unused, and were likely just for reference.

Reply:
Hm I still don't see why they chose `384` rather than `320` for ZOOM_WIDTH.

Reply:
Also in T_DrawZoom `scr_pix_height` is odly `192` and not `224`